src/main/java/cn/nukkit/entity/passive/EntityVillager.java

```diff
@@ -76,60 +98,54 @@ public class EntityVillager extends EntityCreature implements InventoryHolder, E
-                """));
-        var input2 = Item.fromString("minecraft:emerald");
-        input2.setCount(1);
-        var output2 = Item.fromString("minecraft:string");
-        output2.setCount(20);
-        this.recipes.add(RecipeBuildUtils.of(input1, input12, output1)
-                        .setMaxUses(16)
-                        .setPriceMultiplierA(0.05f)
-                        .setRewardExp((byte) 1)
-                        .setTier(1)
-                        .setTraderExp(2)
-                        .build())
-                .add(RecipeBuildUtils.of(input2, output2)
-                        .setMaxUses(16)
-                        .setPriceMultiplierA(0.05f)
-                        .setRewardExp((byte) 1)
-                        .setTier(2)
-                        .setTraderExp(2)
-                        .build())
-                .add(RecipeBuildUtils.of(input1, output1)
-                        .setMaxUses(16)
-                        .setPriceMultiplierA(0.05f)
-                        .setRewardExp((byte) 1)
-                        .setTier(3)
-                        .setTraderExp(2)
-                        .build());
+    {
         this.tierExpRequirement = new int[]{0, 10, 70, 150, 250};
     }
 
     public EntityVillager(FullChunk chunk, CompoundTag nbt) {
         super(chunk, nbt);
     }
 
+    @Override
+    public IBehaviorGroup requireBehaviorGroup() {
+        return new BehaviorGroup(
+                this.tickSpread,
+                Set.of(
+                        //用于刷新InLove状态的核心行为
+                        new Behavior(
+                                new InLoveExecutor(400),
+                                all(
+                                        new PassByTimeEvaluator(CoreMemoryTypes.LAST_BE_FEED_TIME, 0, 400),
+                                        new PassByTimeEvaluator(CoreMemoryTypes.LAST_IN_LOVE_TIME, 6000, Integer.MAX_VALUE)
+                                ),
+                                1, 1
+                        ),
+                        //生长
+                        new Behavior(
+                                new AnimalGrowExecutor(),
+                                //todo：Growth rate
+                                all(
+                                        new PassByTimeEvaluator(CoreMemoryTypes.ENTITY_SPAWN_TIME, 20 * 60 * 20, Integer.MAX_VALUE),
+                                        entity -> entity instanceof EntityAnimal animal && animal.isBaby()
+                                )
+                                , 1, 1, 1200
+                        )
+                ),
+                Set.of(
+                        new Behavior(new FlatRandomRoamExecutor(0.4f, 12, 40, true, 100, true, 10), new PassByTimeEvaluator(CoreMemoryTypes.LAST_BE_ATTACKED_TIME, 0, 100), 4, 1),
+                        new Behavior(new EntityBreedingExecutor<>(EntityPig.class, 16, 100, 0.5f), entity -> entity.getMemoryStorage().get(CoreMemoryTypes.IS_IN_LOVE), 3, 1),
```

Comment:
Why do we need `EntityBreedingExecutor` with `EntityPig.class`? Can we feed the villagers?

Reply:
Uh. I focused on the functionality. I probably missed that